store/sqlstore/post_store.go

```diff
@@ -254,18 +256,24 @@ func (s *SqlPostStore) Update(newPost *model.Post, oldPost *model.Post) (*model.
 	}
 
 	if _, err := s.GetMaster().Update(newPost); err != nil {
-		return nil, model.NewAppError("SqlPostStore.Update", "store.sql_post.update.app_error", nil, "id="+newPost.Id+", "+err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "failed to update Post with id=%s", newPost.Id)
 	}
 
 	time := model.GetMillis()
-	s.GetMaster().Exec("UPDATE Channels SET LastPostAt = :LastPostAt  WHERE Id = :ChannelId AND LastPostAt < :LastPostAt", map[string]interface{}{"LastPostAt": time, "ChannelId": newPost.ChannelId})
+	if _, nErr := s.GetMaster().Exec("UPDATE Channels SET LastPostAt = :LastPostAt  WHERE Id = :ChannelId AND LastPostAt < :LastPostAt", map[string]interface{}{"LastPostAt": time, "ChannelId": newPost.ChannelId}); nErr != nil {
+		return nil, errors.Wrap(nErr, "failed to update Channels")
+	}
 
 	if len(newPost.RootId) > 0 {
-		s.GetMaster().Exec("UPDATE Posts SET UpdateAt = :UpdateAt WHERE Id = :RootId AND UpdateAt < :UpdateAt", map[string]interface{}{"UpdateAt": time, "RootId": newPost.RootId})
+		if _, nErr := s.GetMaster().Exec("UPDATE Posts SET UpdateAt = :UpdateAt WHERE Id = :RootId AND UpdateAt < :UpdateAt", map[string]interface{}{"UpdateAt": time, "RootId": newPost.RootId}); nErr != nil {
+			return nil, errors.Wrap(nErr, "failed to update Posts")
+		}
 	}
 
 	// mark the old post as deleted
-	s.GetMaster().Insert(oldPost)
+	if nErr := s.GetMaster().Insert(oldPost); nErr != nil {
```

Comment:
I'm not 100% sure about this generating a failure. what do you think @agnivade I think here should fail.

Reply:
Same as above, return the error or logging it at least if the return is too much

Reply:
Yes, I think the updating of the newPost and insertion of oldPost should happen in a transaction and we should fail if any one fails. But this PR is already pretty big, I think we can leave that part of changing this to a transaction in a separate PR.

Regarding the other 2, we are not much bothered to check the errors, because like @jespino said, the next post update should fix this. Yes, ideally we should log them. But using a logger at such a low level like store is not really a good practice. The leaf functions should just do computation and return values. Only the top-level functions should be responsible for logging the success/error.

That said, I completely agree that having an unhandled error function in the code is not a solution too. Let me think some more on it. For now, let's keep the old behavior of ignoring errors.

Reply:
Yep, I missed the fact of adding a logger in this layer is not appropiate. I will revert this part of the code because it lacks of transaction so we can check it again in another PR with more calm and time